extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java

```diff
@@ -27,14 +29,21 @@
     @PostConstruct
     protected void init() {
         Set<Bean<?>> beans = Arc.container().beanManager().getBeans(DataSource.class);
+        DataSourcesExcludedFromHealthChecks excluded = Arc.container().instance(DataSourcesExcludedFromHealthChecks.class)
+                .get();
+        List<String> excludedNames = excluded.getExcludedNames();
         for (Bean<?> bean : beans) {
             if (bean.getName() == null) {
-                // this is the default DataSource: retrieve it by type
-                DataSource defaultDs = Arc.container().instance(DataSource.class).get();
-                dataSources.put(DEFAULT_DS, defaultDs);
+                if (!excludedNames.contains(null)) {
```

Comment:
Let's replace the null usage by something like https://github.com/quarkusio/quarkus/blob/e963fe4f4887cdeaf0bdee3ae9f4040be0736a8b/extensions/datasource/common/src/main/java/io/quarkus/datasource/common/runtime/DataSourceUtil.java#L7

Reply:
done